Reduce even more header includes in Frame.h and LocalFrame.h

rniwa opened the pull request:
#### 7ef7df060b4881a4e2f3cc59dcdf61d2dc242199
<pre>
Reduce even more header includes in Frame.h and LocalFrame.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=292707">https://bugs.webkit.org/show_bug.cgi?id=292707</a>

Reviewed by Anne van Kesteren.

This PR replaces more header includes in Frame.h and LocalFrame.h with forward declarations.

In the case of m_overrideScreenSize and m_storageAccessExceptionDomains, made them unique_ptr
to avoid having to #include the relevant headers.

It also extracts ScrollbarMode.h out of ScrollTypes.h to be included in LocalFrame.h.

Finally, it also converts AdjustViewSizeOrNot to an enum class and renames it to AdjustViewSize.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/identity/DigitalCredential.cpp:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/html/ImageDocument.cpp:
* Source/WebCore/html/MediaDocument.cpp:
* Source/WebCore/html/ModelDocument.cpp:
* Source/WebCore/html/OwnerPermissionsPolicyData.h:
* Source/WebCore/html/PDFDocument.cpp:
* Source/WebCore/html/PermissionsPolicy.cpp:
* Source/WebCore/html/PluginDocument.cpp:
* Source/WebCore/html/parser/HTMLConstructionSite.cpp:
* Source/WebCore/page/AdjustViewSize.h: Renamed from Source/WebCore/page/AdjustViewSizeOrNot.h.
* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::setOwnerPermissionsPolicy):
(WebCore::Frame::ownerPermissionsPolicy const):
* Source/WebCore/page/Frame.h:
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::setPrinting):
(WebCore::LocalFrame::screenSize const):
(WebCore::LocalFrame::setOverrideScreenSize):
(WebCore::LocalFrame::storageAccessExceptionReceivedForDomain):
(WebCore::LocalFrame::requestSkipUserActivationCheckForStorageAccess):
* Source/WebCore/page/LocalFrame.h:
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::forceLayoutForPagination):
* Source/WebCore/page/LocalFrameView.h:
* Source/WebCore/page/PrintContext.cpp:
(WebCore::PrintContext::begin):
(WebCore::PrintContext::end):
* Source/WebCore/platform/ScrollTypes.h:
* Source/WebCore/platform/ScrollbarMode.h: Added.
* Source/WebCore/xml/parser/XMLDocumentParserLibxml2.cpp:
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
(-[WebHTMLView layoutToMinimumPageWidth:height:originalPageWidth:originalPageHeight:maximumShrinkRatio:adjustingViewSize:]):

Canonical link: <a href="https://commits.webkit.org/294671@main">https://commits.webkit.org/294671@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/92a6af282284a121288c8ba9e4273f7da3218dc4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/102607 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/22276 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/12590 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/107768 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/53244 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/104646 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/22584 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/30782 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/78050 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/35025 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/105613 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/17497 "Passed tests") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/92603 "Found 1 new API test failure: TestWebKitAPI.WebKit.PrintFrame (failure)") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/58385 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/17336 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/10635 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/52601 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/87147 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/10703 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/110143 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/29739 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/21923 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/87028 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/30103 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/88800 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/86635 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/31458 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/9185 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/23989 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16664 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/29667 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/34979 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/29478 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/32801 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/31037 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/page/AdjustViewSize.h:
```diff
@@ -27,9 +27,9 @@
-    AdjustViewSize
+enum class AdjustViewSize : bool {
+    No,
+    Yes
```
Tiniest of optional nits: usually these "No, Yes" versions go on a single line.

rniwa replied:
Sure, will change.

annevk commented on Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
```diff
@@ -644,9 +644,9 @@
 		A17191AD2CD2DB090088944E /* WKWebViewLogging.mm in Sources */ = {isa = PBXBuildFile; fileRef = A17191AC2CD2DB090088944E /* WKWebViewLogging.mm */; };
-		A176D24B2D6D2A4E00C63915 /* GPUExtension.appex in Embed Extensions */ = {isa = PBXBuildFile; fileRef = A176D2472D6D295400C63915 /* GPUExtension.appex */; platformFilters = (ios, ); settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
-		A176D24C2D6D2A4E00C63915 /* NetworkingExtension.appex in Embed Extensions */ = {isa = PBXBuildFile; fileRef = A176D2482D6D295400C63915 /* NetworkingExtension.appex */; platformFilters = (ios, ); settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
-		A176D24D2D6D2A4E00C63915 /* WebContentExtension.appex in Embed Extensions */ = {isa = PBXBuildFile; fileRef = A176D2492D6D295400C63915 /* WebContentExtension.appex */; platformFilters = (ios, ); settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
```
These changes seem unrelated, but also harmless.

rniwa replied:
I think it's Xcode doing its own thing :(

rniwa replied:
Reverting.

rniwa committed:
Reduce even more header includes in Frame.h and LocalFrame.h
https://bugs.webkit.org/show_bug.cgi?id=292707

Reviewed by Anne van Kesteren.

This PR replaces more header includes in Frame.h and LocalFrame.h with forward declarations.

In the case of m_overrideScreenSize and m_storageAccessExceptionDomains, made them unique_ptr
to avoid having to #include the relevant headers.

It also extracts ScrollbarMode.h out of ScrollTypes.h to be included in LocalFrame.h.

Finally, it also converts AdjustViewSizeOrNot to an enum class and renames it to AdjustViewSize.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/identity/DigitalCredential.cpp:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/html/ImageDocument.cpp:
* Source/WebCore/html/MediaDocument.cpp:
* Source/WebCore/html/ModelDocument.cpp:
* Source/WebCore/html/OwnerPermissionsPolicyData.h:
* Source/WebCore/html/PDFDocument.cpp:
* Source/WebCore/html/PermissionsPolicy.cpp:
* Source/WebCore/html/PluginDocument.cpp:
* Source/WebCore/html/parser/HTMLConstructionSite.cpp:
* Source/WebCore/page/AdjustViewSize.h: Renamed from Source/WebCore/page/AdjustViewSizeOrNot.h.
* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::setOwnerPermissionsPolicy):
(WebCore::Frame::ownerPermissionsPolicy const):
* Source/WebCore/page/Frame.h:
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::setPrinting):
(WebCore::LocalFrame::screenSize const):
(WebCore::LocalFrame::setOverrideScreenSize):
(WebCore::LocalFrame::storageAccessExceptionReceivedForDomain):
(WebCore::LocalFrame::requestSkipUserActivationCheckForStorageAccess):
* Source/WebCore/page/LocalFrame.h:
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::forceLayoutForPagination):
* Source/WebCore/page/LocalFrameView.h:
* Source/WebCore/page/PrintContext.cpp:
(WebCore::PrintContext::begin):
(WebCore::PrintContext::end):
* Source/WebCore/platform/ScrollTypes.h:
* Source/WebCore/platform/ScrollbarMode.h: Added.
* Source/WebCore/xml/parser/XMLDocumentParserLibxml2.cpp:
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
(-[WebHTMLView layoutToMinimumPageWidth:height:originalPageWidth:originalPageHeight:maximumShrinkRatio:adjustingViewSize:]):

Canonical link: https://commits.webkit.org/294671@main
